### DISCUSSION TITLE
Add control panel variables

### DISCUSSION THREAD

**Discussion by ryanleichty** [craftcms/cms#15635] [Ideas]
In Sanity, you build your schemas via code and have full access to JavaScript variables. It would be great if some kind of variables were added to Craft's control panel to further systematize how fields and everything else is managed.

I could imagine these being usable just about everywhere within the control panel to share values. Labels, handles (maybe not?), instructions, character limits, placeholder text, notes, etc.

![CleanShot 2024-08-30 at 19 20 48@2x](https://github.com/user-attachments/assets/9fcb8aa7-df93-4e9c-b599-97d646499efc)

![CleanShot 2024-08-30 at 19 31 30@2x](https://github.com/user-attachments/assets/5030c396-68e9-4889-8421-99a1ca5c29aa)

![CleanShot 2024-08-30 at 19 39 07@2x](https://github.com/user-attachments/assets/979bd3c0-fea9-4b74-982d-3102bdfe6c34)

**Comment by ryanleichty** [comment]
Shopify has a similar kind of UI when referencing `metafields` or `metaobjects`:

![CleanShot 2024-08-30 at 15 54 21@2x](https://github.com/user-attachments/assets/9672dff7-eec0-4e11-a01a-24bcee35537f)

And Figma recently added variables:

![CleanShot 2024-08-30 at 15 57 23@2x](https://github.com/user-attachments/assets/6ffef585-e13f-4cd0-ac79-f4ba73d84db3)

![CleanShot 2024-08-30 at 15 58 06@2x](https://github.com/user-attachments/assets/15fad7ca-e3d2-41c6-80f8-50194abcaf42)

**Reply by thupsi** [reply]
Interesting idea!
